src/basic/chattr-util.h

```diff
@@ -31,6 +31,11 @@
          FS_NOCOW_FL        |                   \
          FS_PROJINHERIT_FL)
 
+/* Project id range for disk quotas */
+#define PROJ_ID_MIN UINT32_C(2147483648)
+#define PROJ_ID_MAX UINT32_C(4294967294)
+#define PROJ_ID_CLAMP_INTO_QUOTA_RANGE(id) ((uint32_t) ((id) % (PROJ_ID_MAX - PROJ_ID_MIN + 1)) + PROJ_ID_MIN)
```

Comment:
these three are not generic project id concepts, but specific to their use for the per-unit quota stuff, hence please move them somewhere into src/core/ I think, and in a commit that actually deals with src/core/ otherwise.